pkg/debug/checkups/notary.go

```diff
@@ -0,0 +1,76 @@
+package checkups
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+type (
+	notaryCheckup struct {
+		k       types.Knapsack
+		status  Status
+		summary string
+		data    map[string]any
+	}
+
+	notaryRelease struct {
+		Signed struct {
+			Version int `json:"version"`
+		} `json:"signed"`
+	}
+)
+
+func (nc *notaryCheckup) Data() map[string]any  { return nc.data }
+func (nc *notaryCheckup) ExtraFileName() string { return "" }
+func (nc *notaryCheckup) Name() string          { return "Notary Version" }
+func (nc *notaryCheckup) Status() Status        { return nc.status }
+func (nc *notaryCheckup) Summary() string       { return nc.summary }
+
+func (nc *notaryCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	nc.data = make(map[string]any)
+	if !nc.k.KolideHosted() || !nc.k.Autoupdate() {
+		return nil
+	}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+	notaryEndpoint := fmt.Sprintf("%s/v2/kolide/launcher/_trust/tuf/targets/releases.json", nc.k.NotaryServerURL())
+	notaryUrl, err := parseUrl(nc.k, notaryEndpoint)
+	if err != nil {
+		return err
+	}
+
+	response, err := fetchNotaryVersion(httpClient, notaryUrl)
```

Comment:
It might be interesting to compare the metadata version for releases on notary.kolide.co with the local tuf repo on disk (same file can be found at `/var/kolide-k2/k2device-preprod.kolide.com/launcher-tuf/targets/releases.json` for darwin/linux -- same dir should be should be in `data` on Windows iirc). That would tell us "does this installation of launcher has the most recent version of notary data available". But maybe more effort than it's worth for the legacy autoupdater.

Reply:
Thinking on it more, I think this falls into "interesting but not worth doing" haha